src/main/java/com/amazon/ion/impl/IonReaderContinuableCoreBinary.java

```diff
@@ -1007,9 +1045,367 @@ private boolean readBoolean_1_1() {
+                        if (valueUnavailable()) {
+                            return;
+                        }
+                        newSymbols.add(stringValue());
+                        state = State.IN_SYMBOL_TABLE_LIST;
+                        break;
+                    case IN_MACRO_TABLE_SEXP:
+                        event = IonReaderContinuableCoreBinary.super.nextValue();
+                        if (event == Event.NEEDS_DATA) {
+                            return;
+                        }
+                        if (event == Event.END_CONTAINER) {
+                            stepOutOfSexpWithinEncodingDirective();
+                            break;
+                        }
+                        if (valueTid.type != IonType.SEXP) {
+                            throw new IonException("macro_table s-expression must contain s-expressions.");
+                        }
+                        state = State.ON_MACRO_SEXP;
+                        break;
+                    case ON_MACRO_SEXP:
+                        if (valueUnavailable()) {
+                            return;
+                        }
+                        state = State.COMPILING_MACRO;
+                        Macro newMacro = macroCompiler.compileMacro();
+                        newMacros.put(MacroRef.byId(++localMacroMaxOffset), newMacro);
+                        state = State.IN_MACRO_TABLE_SEXP;
+                        break;
+                    case COMPILING_MACRO:
+                        // This state can only be reached during compilation of a macro. Do nothing, as the reader must
+                        // navigate normally while the macro is compiled.
+                        break;
+                    default:
+                        throw new IllegalStateException(state.toString());
+                }
+            }
+        }
+
+        void resetState() {
+            isSymbolTableAppend = false;
+            newSymbols.clear();
+        }
+    }
+
+    /**
+     * The reader's state. `READING_VALUE` indicates that the reader is reading a raw value; all other states
+     * indicate that the reader is in the middle of reading an encoding directive.
+     */
+    private enum State {
+        ON_ION_ENCODING_SEXP,
+        IN_ION_ENCODING_SEXP,
+        ON_SEXP_IN_ION_ENCODING,
+        IN_SEXP_IN_ION_ENCODING,
+        CLASSIFYING_SEXP_IN_ION_ENCODING,
+        IN_SYMBOL_TABLE_SEXP,
+        IN_APPENDED_SYMBOL_TABLE,
+        ON_SYMBOL_TABLE_LIST,
+        IN_SYMBOL_TABLE_LIST,
+        ON_SYMBOL,
+        IN_MACRO_TABLE_SEXP,
+        ON_MACRO_SEXP,
+        COMPILING_MACRO,
+        READING_VALUE,
+    }
+
+    // The current state.
+    private State state = State.READING_VALUE;
+
+    /**
+     * Navigates to the next raw Ion 1.1 value, consuming any encoding directives that occur between raw values.
+     * @return an event conveying the result of the operation.
+     */
+    private Event nextValue_1_1() {
```

Comment:
Note: this looks very similar to the application-level's nextValue() implementation.